test/integration/framework/api.go

```diff
@@ -96,24 +96,27 @@ func (f *FederationAPIFixture) NewClient(userAgent string) federationclientset.I
 }
 
 func startServer(t *testing.T, runOptions *options.ServerRunOptions, stopChan <-chan struct{}) error {
+	var port int
 	err := wait.PollImmediate(DefaultWaitInterval, wait.ForeverTestTimeout, func() (bool, error) {
-		port, err := framework.FindFreeLocalPort()
+		var err error
+		port, err = framework.FindFreeLocalPort()
 		if err != nil {
 			t.Logf("Error allocating an ephemeral port: %v", err)
 			return false, nil
 		}
-
-		runOptions.InsecureServing.BindPort = port
-		err = app.NonBlockingRun(runOptions, stopChan)
-		if err != nil {
-			t.Logf("Error starting the %s: %v", apiNoun, err)
-			return false, nil
-		}
 		return true, nil
 	})
 	if err != nil {
 		return fmt.Errorf("Timed out waiting for the %s: %v", apiNoun, err)
 	}
+
+	runOptions.InsecureServing.BindPort = port
+	runOptions.SecureServing.ServerCert.CertDirectory = "/tmp"
```

Comment:
Will this be created if `/tmp` does not exist or not mounted? `os` packages TmpDir() can be used or taken as an example.

Reply:
no "/tmp" will not be created but only used here. This is kind of test fixture, /tmp will be available on most root file systems and is meant for temporary usage which is world writable. I just did the simplest thing necessary, to fix the issue.

Reply:
okie, updated.